File: bddtests/src/main/kotlin/mocking/emis/models/MedicationCourse.kt
package mocking.emis.models

data class MedicationCourse(
        val medicationCourseGuid : String,
        val name: String,
        val dosage: String?,
        val quantityRepresentation: String?,
        var prescriptionType: PrescriptionType,
        val constituents: List<String>,
        var canBeRequested: Boolean)

{
    fun getInstructionsText() : String {
        return pages.prescription.resolveDetailsField(dosage, quantityRepresentation)
    }
}

File: xamarinintegrationtests/NHSOnline.HttpMocks/Emis/Models/Prescriptions/MedicationCourse.cs
using System.Collections.Generic;
using System.Diagnostics.CodeAnalysis;

namespace NHSOnline.HttpMocks.Emis.Models.Prescriptions
{
    [SuppressMessage("Usage", "CA2227:Collection properties should be read only",
        Justification = "Required for mocks")]
    public class MedicationCourse
    {
        public string? MedicationCourseGuid { get; set; }
        public string? Name { get; set; }
        public string? Dosage { get; set; }
        public string? QuantityRepresentation { get; set; }
        public PrescriptionType? PrescriptionType { get; set; }
        public IList<string>? Constituents { get; set; }
        public bool CanBeRequested { get; set; }
    }
}